ocaml/xapi/storage_smapiv1_migrate.ml

```diff
@@ -563,16 +567,26 @@ let mirror_cleanup ~dbg ~sr ~snapshot =
 module MIRROR : SMAPIv2_MIRROR = struct
   type context = unit
 
+  let u x = raise Storage_interface.(Storage_error (Errors.Unimplemented x))
+
   let send_start _ctx ~dbg ~task_id ~dp ~sr ~vdi ~mirror_vm ~mirror_id
-      ~local_vdi ~copy_vm ~live_vm:_ ~url ~remote_mirror ~dest_sr ~verify_dest =
+      ~local_vdi ~copy_vm ~live_vm ~url ~remote_mirror ~dest_sr ~verify_dest =
     let (module Remote) =
       Storage_migrate_helper.get_remote_backend url verify_dest
     in
     match remote_mirror with
+    | Mirror.SMAPIv3_mirror _ ->
+        (* this should never happen *)
+        raise
+          (Storage_error
+             (Migration_mirror_failure
+                "Incorrect remote mirror format for SMAPIv1"
```

Comment:
Could more information be included to help diagnosis when this happens?

Reply:
This is a "this should never happen" situation and is only there to make the compiler happy... I don't think user should encounter this error in any case, and there is probably nothing they can do about it if they do for some reason...